datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1145,6 +1145,22 @@ impl Projection {
         })
     }
 
+    /// Create a new Projection using the specified output schema
+    pub fn new_from_schema(input: Arc<LogicalPlan>, schema: DFSchemaRef) -> Self {
+        let expr: Vec<Expr> = schema
```

Comment:
Do not see a place this method is used. I think if we have logic to reorder joins, we might need to add an additional Projection if the output schema is not the same with the original plan.